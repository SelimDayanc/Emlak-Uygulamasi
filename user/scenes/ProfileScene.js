import React, {Component, PropTypes} from 'react';
import {StyleSheet, View} from 'react-native';
import UserInfo from '../components/profile/UserInfo';
import UserLogo from '../components/profile/UserLogo';
import Contact from '../components/profile/Contact';
import PropertyListScene from '../../property/scenes/PropertyListScene';
import {TabBar, TabViewAnimated} from 'react-native-tab-view';
import colors from '../../common/colors';
import {CountryPropType} from '../../property/common/proptypes';

export default class ProfileScene extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
    properties: PropTypes.array.isRequired,
    isFetching: PropTypes.bool.isRequired,
    loadScene: PropTypes.func.isRequired,
    handleFavoritePress: PropTypes.func.isRequired,
    fetchProperties: PropTypes.func.isRequired,
    country: CountryPropType.isRequired,
  };

  state = {
    index: 0,
    routes: [
      {key: '1', title: 'Basic Info'},
      {key: '2', title: 'Properties'},
      {key: '3', title: 'Contact'},
    ],
  };

  handleChangeTab = index => {
    this.setState({index});
  };

  renderHeader = props => {
    return (
      <TabBar
        {...props}
        scrollEnabled
        indicatorStyle={styles.indicator}
        style={styles.tabbar}
        labelStyle={styles.label}
      />
    );
  };

  renderScene = ({route}) => {
    const {
      user,
      properties,
      isFetching,
      loadScene,
      handleFavoritePress,
      fetchProperties,
      country,
      refreshProperties,
    } = this.props;
    switch (route.key) {
      case '1':
        return <UserInfo user={user} />;
      case '2':
        return (
          <PropertyListScene
            collection={properties}
            loadScene={loadScene}
            handleFavoritePress={handleFavoritePress}
            isFetching={isFetching}
            fetchProperties={fetchProperties}
            country={country}
            refreshProperties={refreshProperties}
          />
        );
      case '3':
        return <Contact user={user} />;
      default:
        return null;
    }
  };

  render() {
    const {user} = this.props;
    return (
      <View style={{flex: 1}}>
        <UserLogo user={user} />
        <TabViewAnimated
          style={styles.container}
          navigationState={this.state}
          renderScene={this.renderScene}
          renderHeader={this.renderHeader}
          onRequestChangeTab={this.handleChangeTab}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  tabbar: {
    backgroundColor: 'white',
  },
  indicator: {
    height: 2,
    backgroundColor: colors.accent,
  },
  label: {
    color: colors.smokeGreyDark,
    fontWeight: '400',
  },
});
