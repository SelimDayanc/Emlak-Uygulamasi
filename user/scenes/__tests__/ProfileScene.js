import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProfileScene from '../../scenes/ProfileScene';

test('renders Profile Scene', () => {
  const tree = renderer
    .create(
      <ProfileScene
        user={{
          name: 'name',
          email: 'z4ls@live.com',
          isCompany: true,
          company: {description: 'company description'},
        }}
        properties={[]}
        isFetching={false}
        loadScene={() => {}}
        handleFavoritePress={() => {}}
        fetchProperties={() => {}}
        country={{
          fullName: 'Kuwait',
          abbr: 'KW',
          currency: 'KD',
          coords: {
            latitude: 29.3759,
            longitude: 47.9774,
          },
        }}
        refreshProperties={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
