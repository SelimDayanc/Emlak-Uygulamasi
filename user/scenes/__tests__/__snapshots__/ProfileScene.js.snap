// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile Scene 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 200,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï€¾
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#484848",
          "fontSize": 20,
          "fontWeight": "700",
          "paddingTop": 5,
        }
      }
    >
      name
    </Text>
  </View>
  <View
    configureTransition={[Function]}
    initialLayout={
      Object {
        "height": 0,
        "width": 0,
      }
    }
    loaded={
      Array [
        0,
      ]
    }
    navigationState={
      Object {
        "index": 0,
        "routes": Array [
          Object {
            "key": "1",
            "title": "Basic Info",
          },
          Object {
            "key": "2",
            "title": "Properties",
          },
          Object {
            "key": "3",
            "title": "Contact",
          },
        ],
      }
    }
    onChangePosition={[Function]}
    onLayout={[Function]}
    onRequestChangeTab={[Function]}
    render={[Function]}
    renderHeader={[Function]}
    renderPager={[Function]}
    renderScene={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "elevation": 4,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0.5,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 0.5,
            "zIndex": 1,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": -0,
                },
              ],
              "width": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#ff1144",
                "bottom": 0,
                "height": 2,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "overflow": "scroll",
            }
          }
        >
          <RCTScrollView
            alwaysBounceHorizontal={false}
            alwaysBounceVertical={false}
            automaticallyAdjustContentInsets={false}
            bounces={false}
            contentContainerStyle={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexWrap": "nowrap",
                },
                null,
              ]
            }
            contentOffset={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            horizontal={true}
            onContentSizeChange={null}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={undefined}
            onResponderTerminationRequest={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onScrollShouldSetResponder={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            scrollEnabled={true}
            scrollEventThrottle={16}
            scrollsToTop={false}
            sendMomentumEvents={true}
            showsHorizontalScrollIndicator={false}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
          >
            <RCTScrollContentView
              collapsable={false}
              removeClippedSubviews={undefined}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  Array [
                    Object {
                      "flexDirection": "row",
                      "flexWrap": "nowrap",
                    },
                    null,
                  ],
                ]
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel="Basic Info"
                accessibilityTraits="button"
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "opacity": 1,
                        "padding": 8,
                        "width": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "white",
                            "margin": 8,
                          },
                          Object {
                            "color": "#5d5f63",
                            "fontWeight": "400",
                          },
                        ]
                      }
                    >
                      BASIC INFO
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel="Properties"
                accessibilityTraits="button"
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "opacity": 0.7,
                        "padding": 8,
                        "width": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "white",
                            "margin": 8,
                          },
                          Object {
                            "color": "#5d5f63",
                            "fontWeight": "400",
                          },
                        ]
                      }
                    >
                      PROPERTIES
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel="Contact"
                accessibilityTraits="button"
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "opacity": 0.7,
                        "padding": 8,
                        "width": 0,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "white",
                            "margin": 8,
                          },
                          Object {
                            "color": "#5d5f63",
                            "fontWeight": "400",
                          },
                        ]
                      }
                    >
                      CONTACT
                    </Text>
                  </View>
                </View>
              </View>
            </RCTScrollContentView>
          </RCTScrollView>
        </View>
      </View>
      <RCTScrollView
        alwaysBounceHorizontal={false}
        alwaysBounceVertical={false}
        automaticallyAdjustContentInsets={false}
        bounces={false}
        contentContainerStyle={
          Object {
            "flexGrow": 1,
          }
        }
        contentOffset={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        directionalLockEnabled={true}
        horizontal={true}
        keyboardDismissMode="on-drag"
        keyboardShouldPersistTaps="always"
        onContentSizeChange={null}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={undefined}
        onResponderTerminationRequest={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onScrollShouldSetResponder={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        pagingEnabled={true}
        scrollEnabled={undefined}
        scrollEventThrottle={16}
        scrollsToTop={false}
        sendMomentumEvents={true}
        showsHorizontalScrollIndicator={false}
        style={
          Array [
            Object {
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "overflow": "scroll",
            },
            Object {
              "flexGrow": 1,
            },
          ]
        }
      >
        <RCTScrollContentView
          collapsable={false}
          removeClippedSubviews={undefined}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "flexGrow": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "overflow": "hidden",
              }
            }
            testID={undefined}
          >
            <RCTScrollView
              alwaysBounceHorizontal={undefined}
              alwaysBounceVertical={true}
              onContentSizeChange={null}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={undefined}
              onResponderTerminationRequest={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              onScrollShouldSetResponder={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              sendMomentumEvents={false}
              style={
                Array [
                  Object {
                    "flexDirection": "column",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "overflow": "scroll",
                  },
                  Object {
                    "backgroundColor": "white",
                    "flex": 1,
                    "padding": 10,
                    "paddingTop": 20,
                  },
                ]
              }
            >
              <RCTScrollContentView
                collapsable={false}
                removeClippedSubviews={undefined}
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                <View
                  style={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#484848",
                        "fontSize": 15,
                        "fontWeight": "100",
                      }
                    }
                  >
                    company description
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#e3e7ed",
                          "height": 0.5,
                        },
                        Object {
                          "marginVertical": 20,
                        },
                      ]
                    }
                  />
                </View>
              </RCTScrollContentView>
            </RCTScrollView>
          </View>
          <View
            style={null}
            testID={undefined}
          />
          <View
            style={null}
            testID={undefined}
          />
        </RCTScrollContentView>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
